### ISSUE TITLE
Analyzer CA1822 (static member) doesn't respect interface properties

### ISSUE THREAD

**Issue by jeremy-morren** [dotnet/roslyn-analyzers#7088]
For a class that implements a property of an interface defined in a different project that doesn't access instance data, analyzer CA1822 will incorrectly flag that it can be made static.

```
//Defined in different project
interface IInterface
{
    string Property { get; }
}

class MyClass : IInterface
{
    //CA1822 warning
    public string Property => string.Empty;
}
```

**Comment by sharwell** [comment]
Is there any way you can provide a complete example for this? I wasn't able to reproduce it with this test:

```cs
[Fact]
public async Task InterfaceImplementationInAnotherProject()
{
    await new VerifyCS.Test
    {
        TestState =
        {
            Sources =
            {
                """
                class MyClass : IInterface
                {
                    public string Property => string.Empty;
                }
                """,
            },
            AdditionalProjects =
            {
                ["Definition"] =
                {
                    Sources =
                    {
                        """
                        public interface IInterface
                        {
                            string Property { get; }
                        }
                        """
                    },
                },
            },
            AdditionalProjectReferences = { "Definition" },
        },
    }.RunAsync();
}
```

**Comment by DavidBetteridge** [comment]
I can recreate the above example,  but only when I add the interface into my existing code base,  so there is something else involved as well.   ( If I create a new solution with two projects I can't get it to happen. )


I do however agree it only seems to happen when the interface is defined in a separate project.


```
public interface IInterface
{
    string P { get; }
}


public class X : IInterface
{
    public string P => string.Empty;
}
```

**Mark members as static**

**Comment by scott-shrout** [comment]
Is this ever going to be addressed or are we going to keep ignoring it?
